src/static_canvas.class.js

```diff
@@ -893,46 +881,32 @@
     /**
      * @private
      * @param {CanvasRenderingContext2D} ctx Context to render on
-     * @param {fabric.Group} activeGroup
+     * @param {Array} objects to render
      */
-    _renderObjects: function(ctx, activeGroup) {
-      var i, length;
-
-      // fast path
-      if (!activeGroup || this.preserveObjectStacking) {
-        for (i = 0, length = this._objects.length; i < length; ++i) {
-          this._draw(ctx, this._objects[i]);
-        }
-      }
-      else {
-        for (i = 0, length = this._objects.length; i < length; ++i) {
-          if (this._objects[i] && !activeGroup.contains(this._objects[i])) {
-            this._draw(ctx, this._objects[i]);
-          }
-        }
+    _renderObjects: function(ctx, objects) {
+      for (var i = 0, length = objects.length; i < length; ++i) {
+        objects[i] && objects[i].render(ctx);
       }
     },
 
     /**
      * @private
      * @param {CanvasRenderingContext2D} ctx Context to render on
-     * @param {fabric.Group} activeGroup
      */
-    _renderActiveGroup: function(ctx, activeGroup) {
+    _renderBackgroundOrOverlay: function(ctx, property) {
+      if (this[property + 'Color']) {
```

Comment:
That would be great! And thanks so much for this effort and your patience ;)

On Mon, Oct 12, 2015 at 1:39 PM, Andrea Bogazzi <notifications@github.com>
wrote:

> In src/static_canvas.class.js
> <https://github.com/kangax/fabric.js/pull/2405#discussion_r41781131>:
>
> >       */
> > -    _renderActiveGroup: function(ctx, activeGroup) {
> > +    _renderBackgroundOrOverlay: function(ctx, property) {
> > +      if (this[property + 'Color']) {
>
> i can make a resubmit solving your comments if you are interested in a
> merge. as of now is not mergeable. Il 12/ott/2015 19:24, "Juriy Zaytsev" <
> notifications@github.com> ha scritto:
> … <#1505d230f4fa9bb4_>
> In src/static_canvas.class.js <#2405 (comment)
> <https://github.com/kangax/fabric.js/pull/2405#discussion_r41779698>>: >
> */ > - _renderActiveGroup: function(ctx, activeGroup) { > +
> _renderBackgroundOrOverlay: function(ctx, property) { > + if (this[property
> + 'Color']) { I wonder if we should perhaps save property + 'Color' to a
> var (to avoid constant concatenation) since this is an often-called
> function, IIRC — Reply to this email directly or view it on GitHub <
> https://github.com/kangax/fabric.js/pull/2405/files#r41779698>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kangax/fabric.js/pull/2405/files#r41781131>.
>